### PR TITLE
Rpc: add getMultipleAccounts endpoint

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
To verify transaction operations and atomicity, it would be helpful for users to be able to query multiple accounts at the same moment in time (same bank).

#### Summary of Changes
- Add `getMultipleAccounts` rpc endpoint. It supports the same configuration options as `getAccountInfo`, but returns results in an indexed array matching the input array.

The input limit is WIP. It is currently set to 20 pubkeys/accounts.

### CODE REVIEW

File: core/src/rpc.rs [hunk 1]
```diff
@@ -270,6 +270,53 @@ impl JsonRpcRequestProcessor {
...
+    pub fn get_multiple_accounts(
+        &self,
+        pubkeys: Vec<Pubkey>,
+        config: Option<RpcAccountInfoConfig>,
+    ) -> Result<RpcResponse<Vec<Option<UiAccount>>>> {
+        let mut accounts: Vec<Option<UiAccount>> = vec![];
+
+        let config = config.unwrap_or_default();
+        let bank = self.bank(config.commitment);
+        let encoding = config.encoding.unwrap_or(UiAccountEncoding::Binary);
```
Comment (mvines) [inline core/src/rpc.rs:282]: For new methods wdyt about moving to Base64 as the default?  It's inconsistent but also the direction that we want to go in I think (that is, any direction away from base58)

File: core/src/rpc.rs [hunk 2]
```diff
@@ -270,6 +270,53 @@ impl JsonRpcRequestProcessor {
...
+        check_slice_and_encoding(&encoding, config.data_slice.is_some())?;
+
+        for pubkey in pubkeys {
+            let mut response_account = None;
+            if let Some(account) = bank.get_account(&pubkey) {
```
Comment (mvines) [inline core/src/rpc.rs:287]: Most of this code can't be shared with `get_account_info()`?

File: core/src/rpc.rs [hunk 3]
```diff
@@ -1766,6 +1821,29 @@ impl RpcSol for RpcSolImpl {
...
+        &self,
+        meta: Self::Metadata,
+        pubkey_strs: Vec<String>,
+        config: Option<RpcAccountInfoConfig>,
+    ) -> Result<RpcResponse<Vec<Option<UiAccount>>>> {
+        debug!(
+            "get_multiple_accounts rpc request received: {:?}",
+            pubkey_strs.len()
+        );
+        if pubkey_strs.len() > NUM_LARGEST_ACCOUNTS {
```
Comment (mvines) [inline core/src/rpc.rs:1834]: Let's not reuse `NUM_LARGEST_ACCOUNTS`.  I don't think somebody would expect changing that constant to affect this method.  A new `MAX_MULTIPLE_ACCOUNTS` perhaps?

**Review by mvines — COMMENTED** [review]
Looks great overall.  Would be good to get confirmation that this'll do the trick.

Also I think you should add a RpcClient method for this please, to flesh out the Rust client side

File: core/src/rpc.rs (hunk 1 above)
Comment (CriesofCarrots) [reply core/src/rpc.rs:282]: I had the same thought as I was typing that Binary. Yeah, let's do it here.

File: core/src/rpc.rs (hunk 3 above)
Comment (CriesofCarrots) [reply core/src/rpc.rs:1834]: No, definitely not. Sorry that wasn't clear. I just threw that in as a starting point, until I get more information about that req.

File: core/src/rpc.rs (hunk 2 above)
Comment (CriesofCarrots) [reply core/src/rpc.rs]: 😳 lazy Tyera

**Comment by CriesofCarrots** [comment]
@mvines I'm going to merge this as-is, and pr configurable `MAX_MULTIPLE_ACCOUNTS` separately

**Pull request merged**